test/jibu/backup_test.go

```diff
@@ -147,32 +160,62 @@ var _ = Describe("use jibu api", func() {
 				Expect(err).ShouldNot(HaveOccurred())
 				MyBy(fmt.Sprintf("backup plan %s created", backupPlan.Metadata.Name))
 
-				MyBy("create a backup job")
-				meta = swagger.V1ObjectMeta{
-					Name: backupJobName,
-				}
-				backupJobSpec := swagger.V1alpha1BackupJobSpec{
-					Action:      actionStartJob,
-					BackupName:  backupPlanName,
-					Desc:        backupJobName,
-					DisplayName: backupJobName,
-					Tenant:      tenant,
-				}
-				backupJob := swagger.V1alpha1BackupJob{
-					Metadata: &meta,
-					Spec:     &backupJobSpec,
+				if !backupRepeatEnabled {
+					MyBy("create a backup job")
+					meta = swagger.V1ObjectMeta{
+						Name: backupJobName,
+					}
+					backupJobSpec := swagger.V1alpha1BackupJobSpec{
+						Action:      actionStartJob,
+						BackupName:  backupPlanName,
+						Desc:        backupJobName,
+						DisplayName: backupJobName,
+						Tenant:      tenant,
+					}
+					backupJob := swagger.V1alpha1BackupJob{
+						Metadata: &meta,
+						Spec:     &backupJobSpec,
+					}
+					_, _, err = jibuClient.BackupJobTagApi.CreateBackupJob(ctx, tenant, backupJob)
+					Expect(err).ShouldNot(HaveOccurred())
+					MyBy(fmt.Sprintf("backup job %s created", backupJobName))
+
+					MyBy(fmt.Sprintf("backup plan should be ready in %v", backupPlanReadyTimeout))
+					waitBackupPlanReady(jibuClient, tenant, backupPlanName)
+					MyBy("back plan is ready now")
+
+					MyBy(fmt.Sprintf("backup job should complete in %v", backupJobFinishedTimeout))
+					waitBackupJobComplete(jibuClient, tenant, backupJobName)
+					MyBy("backup job succeeded")
+				} else {
+					MyBy("wait for repeated creation of backup jobs")
+
+					// indexChan dispatches the index of each backup job
+					// sorted by creation time
+					// which is retrieved and used by each cronjob execution
+					// to find out which job they should be waiting for
+					indexChan := make(chan int, backupRepeatCheckNum)
+					wg := sync.WaitGroup{}
+					for i := 0; i < backupRepeatCheckNum; i++ {
+						indexChan <- i
+						wg.Add(1)
+					}
+					checkRepeatedBackupJob := func() {
+						defer wg.Done()
+						index := <-indexChan
+						MyBy(fmt.Sprintf("wait for backup job to be created in %v, index: %d", backupJobRepeatedCreationTimeout, index))
+						jobName := waitNthBackupJobCreation(jibuClient, tenant, backupPlanName, index)
+						MyBy(fmt.Sprintf("backup job created, index: %d, name: %s", index, jobName))
+						MyBy(fmt.Sprintf("backup job should complete in %v, index: %d, name: %s", backupJobFinishedTimeout, index, jobName))
```

Comment:
question: if every 3 minutes generate a backup job, but each backup job needs take 1 hour to complete, what would happen? will the second backup job be generated 3 minutes later or 1 hour later?

Reply:
After some real experiments,  it turns out that the second backup job will only be created *after* the first backup job is compeleted.

That said, if a backup job stucks, then no backup job will be created as the configured cron schedule.

Reply:
So now I think the current implementation of cron-scheduled checking might not work if the backup job takes a long time to finish: if the first backup job has not finished yet, but the second checking cron job is triggered, it will timeout waiting for the second backup job to be created.

I'll change it to fit into this scenario.